metrics-tracing-context/tests/integration.rs

```diff
@@ -110,7 +120,200 @@ fn test_basic_functionality() {
+        snapshots.push((
+            CompositeKey::new(
+                MetricKind::Counter,
+                Key::from_static_parts(LOGIN_ATTEMPTS, SVC_USER_EMAIL),
+            ),
+            None,
+            None,
+            DebugValue::Counter(1),
+        ));
+
+        assert_eq!(snapshot, snapshots);
+    }
+    span.record("user.id", 42);
+    {
+        counter!("login_attempts", "service" => "login_service").increment(1);
+
+        let snapshot = snapshotter.snapshot().into_vec();
+
+        snapshots.push((
+            CompositeKey::new(
+                MetricKind::Counter,
+                Key::from_static_parts(LOGIN_ATTEMPTS, SVC_USER_EMAIL_ID),
+            ),
+            None,
+            None,
+            DebugValue::Counter(1),
+        ));
+
+        assert_eq!(snapshot, snapshots);
+    }
+}
+
+#[test]
+fn test_rerecord() {
+    static USER_ID_42: &[Label] = &[Label::from_static_parts("user.id", "42")];
+    static USER_ID_123: &[Label] = &[Label::from_static_parts("user.id", "123")];
+
+    let (_guard, snapshotter) = setup(TracingContextLayer::all());
+
+    let span = span!(Level::TRACE, "login", user.id = tracing_core::field::Empty);
+    let _guard = span.enter();
+
+    span.record("user.id", 42);
+    counter!("login_attempts").increment(1);
+
+    span.record("user.id", 123);
+    counter!("login_attempts").increment(1);
+
+    let snapshot = snapshotter.snapshot().into_vec();
+
+    assert_eq!(
+        snapshot,
+        vec![
+            (
+                CompositeKey::new(
+                    MetricKind::Counter,
+                    Key::from_static_parts(LOGIN_ATTEMPTS, USER_ID_42)
+                ),
+                None,
+                None,
+                DebugValue::Counter(1),
+            ),
+            (
+                CompositeKey::new(
+                    MetricKind::Counter,
+                    Key::from_static_parts(LOGIN_ATTEMPTS, USER_ID_123)
+                ),
+                None,
+                None,
+                DebugValue::Counter(1),
+            )
+        ]
+    );
+}
+
+#[test]
+fn test_loop() {
+    let (_guard, snapshotter) = setup(TracingContextLayer::all());
+
+    let user = "ferris";
+    let email = "ferris@rust-lang.org";
+    let span = span!(
+        Level::TRACE,
+        "login",
+        user,
+        user.email = email,
+        attempt = tracing_core::field::Empty,
+    );
+    let _guard = span.enter();
+
+    for attempt in 1..=42 {
+        span.record("attempt", attempt);
+    }
+    counter!("login_attempts").increment(1);
+
+    let snapshot = snapshotter.snapshot().into_vec();
+
+    assert_eq!(
+        snapshot,
+        vec![(
+            CompositeKey::new(
+                MetricKind::Counter,
+                Key::from_static_parts(LOGIN_ATTEMPTS, USER_EMAIL_ATTEMPT)
+            ),
+            None,
+            None,
+            DebugValue::Counter(1),
+        )]
+    );
+}
+
+#[test]
+fn test_record_does_not_overwrite() {
```

Comment:
I think we also need to add a test like this one, but where the overwriting is happening due to a nested parent/child span, showing that the field value in the child span doesn't get overridden by the same field (with a different value) in the parent span.

Reply:
`test_nested_spans` already tests this behavior.

Reply:
Ah yeah, weird... not sure how I missed that. 🤔